test/testframe.c

```diff
@@ -11,368 +11,504 @@
+#include "testframe.h"
+#include "h5test.h"
 
 /*
  * Definitions for the testing structure.
  */
-#define MAXTESTNAME 16
-#define MAXTESTDESC 64
 
 typedef struct TestStruct {
-    int  NumErrors;
-    char Description[MAXTESTDESC];
-    int  SkipFlag;
     char Name[MAXTESTNAME];
-    void (*Call)(void);
-    void (*Cleanup)(void);
-    const void *Parameters;
+    char Description[MAXTESTDESC];
+    void (*TestFunc)(const void *);
+    void (*TestSetupFunc)(void *);
+    void (*TestCleanupFunc)(void *);
+    void *TestParameters;
+    int   TestNumErrors;
+    int   TestSkipFlag;
 } TestStruct;
 
 /*
- * Variables used by testing framework.
+ * Global variables used by testing framework.
  */
-static int         enable_error_stack               = 0;         /* enable error stack; disable=0 enable=1 */
-static int         num_errs                         = 0;         /* Total number of errors during testing */
-int                TestVerbosity                    = VERBO_DEF; /* Default Verbosity is Low */
-static int         Summary                          = 0;         /* Show test summary. Default is no. */
-static int         CleanUp                          = 1;         /* Do cleanup or not. Default is yes. */
-int                TestExpress                      = -1;   /* Do TestExpress or not. -1 means not set yet. */
-static TestStruct *Test                             = NULL; /* Array of tests */
-static unsigned    TestAlloc                        = 0;    /* Size of the Test array */
-static unsigned    Index                            = 0;
-static const void *Test_parameters                  = NULL;
-static const char *TestProgName                     = NULL;
-static void (*TestPrivateUsage)(void)               = NULL;
-static int (*TestPrivateParser)(int ac, char *av[]) = NULL;
-int mpi_rank_framework_g                            = 0;
+
+static TestStruct *TestArray = NULL; /* Array of tests */
+static unsigned    TestAlloc = 0;    /* Size of the Test array */
+static unsigned    TestCount = 0;    /* Number of tests currently added to test array */
+
+static const char *TestProgName                              = NULL;
+static void (*TestPrivateUsage_g)(FILE *stream)              = NULL;
+static herr_t (*TestPrivateParser_g)(int argc, char *argv[]) = NULL;
+
+static int  TestNumErrs_g        = 0;    /* Total number of errors that occurred for whole test program */
+static bool TestEnableErrorStack = true; /* Whether to show error stacks from the library */
+
+static int TestMaxNumThreads_g = -1; /* Max number of threads that can be spawned */
+
+static bool TestDoSummary_g = false; /* Show test summary. Default is no. */
+static bool TestDoCleanUp_g = true;  /* Do cleanup or not. Default is yes. */
+
+int TestFrameworkProcessID_g = 0;         /* MPI process rank value for parallel tests */
+int TestVerbosity_g          = VERBO_DEF; /* Default Verbosity is Low */
 
 /*
- * Setup a test function and add it to the list of tests.
```

Comment:
Documentation for all the functions here went into testframe.h, so header comments became brief.